router/src/server/http.rs

```diff
@@ -385,6 +413,36 @@ where
         let namespace = org_and_bucket_to_namespace(&write_info.org, &write_info.bucket)
             .map_err(OrgBucketError::MappingFail)?;
 
+        if let Some(authz) = &self.authz {
+            let token = req
+                .headers()
+                .get(hyper::header::AUTHORIZATION)
+                .and_then(|v| {
+                    let s = v.as_ref();
+                    if s.len() < b"Token ".len() {
+                        None
+                    } else {
+                        match s.split_at(b"Token ".len()) {
+                            (b"Token ", token) => Some(token),
+                            _ => None,
+                        }
+                    }
+                })
+                .ok_or(Error::Unauthenticated)?;
+
+            let perms = vec![Permission::ResourceAction(
```

Comment:
Minor, but you can drop the `vec!` here:


```suggestion
            let perms = [Permission::ResourceAction(
```

The auth call passes `perms` as a slice, so the old code would allocate a vector (an array with a length), dereference it to a slice, and pass a reference to that slice. 

This change cuts out the middle man, allocates an immutable array, and gives it directly.